### PR TITLE
Fixes the "name 'py_proto_library' is not defined" error.

### PR DESCRIPTION
**Pull request by ruomingp**
Background: https://github.com/tensorflow/lingvo/issues/274#issuecomment-1071115444.

Also adds 'flax' as a pip dependency. Background: https://github.com/tensorflow/lingvo/issues/275.

### CODE REVIEW

**Comment by drpngx** [comment]
Oh, I don't have approve rights either. @rohan-anil ?

File: lingvo/jax/BUILD [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Description:
 #   Base directory for the lingvo Jax library.
 
-load("//lingvo:lingvo.bzl", "pytype_strict_binary", "pytype_strict_library")
+load("//lingvo:lingvo.bzl", "pytype_strict_binary", "pytype_strict_library", "lingvo_proto_py")
```
Comment (laurentes) [inline lingvo/jax/BUILD:4]: Please make this a separate line/instruction because of internal rule rewrite:
load("//lingvo:lingvo.bzl", "lingvo_proto_py")

File: docker/dev.dockerfile [hunk 2]
```diff
@@ -75,6 +75,7 @@ ARG pip_dependencies=' \
       attrs \
       contextlib2 \
       dataclasses \
+      flax \
```
Comment (laurentes) [inline docker/dev.dockerfile:78]: For historical reason, there is a separate docker file for the lingvo+JAX here.
https://github.com/tensorflow/lingvo/blob/master/lingvo/jax/pip_package/build.Dockerfile

Could you revert this change and try to use this docker file instead?

**Review by laurentes — CHANGES_REQUESTED** [review]
Thanks Ruoming!

I'm submitting a change internally to fix the issue with the Flax BUILD dependencies.

File: lingvo/jax/BUILD (hunk 1 above)
Comment (ruomingp) [reply lingvo/jax/BUILD:4]: Thanks. Done.

File: docker/dev.dockerfile (hunk 2 above)
Comment (ruomingp) [reply docker/dev.dockerfile]: Done. 

I also ran into a problem with "//third_party/py/einshape" in jax/layers/BUILD.

Separately, I tried to build docker and ran into a different error:
```
% docker build --tag tensorflow:lingvo $(test "$LINGVO_DEVICE" = "gpu" && echo "--build-arg base_image=nvidia/cuda:11.0-cudnn8-runtime-ubuntu18.04") - < "$LINGVO_DIR/lingvo/jax/pip_package/build.Dockerfile"
...
#11 7.361 E: Unable to locate package python3.7
#11 7.361 E: Couldn't find any package by glob 'python3.7'
#11 7.361 E: Couldn't find any package by regex 'python3.7'
#11 7.361 E: Unable to locate package python3.7-distutils
#11 7.361 E: Couldn't find any package by glob 'python3.7-distutils'
#11 7.361 E: Couldn't find any package by regex 'python3.7-distutils'
#11 7.361 E: Unable to locate package python3.8
#11 7.361 E: Couldn't find any package by glob 'python3.8'
#11 7.361 E: Couldn't find any package by regex 'python3.8'
#11 7.361 E: Unable to locate package python3.8-distutils
#11 7.361 E: Couldn't find any package by glob 'python3.8-distutils'
#11 7.361 E: Couldn't find any package by regex 'python3.8-distutils'
#11 7.361 E: Unable to locate package python3.9-distutils
#11 7.361 E: Couldn't find any package by glob 'python3.9-distutils'
#11 7.361 E: Couldn't find any package by regex 'python3.9-distutils'
------
executor failed running [/bin/sh -c apt-get update && apt-get install -y python3.7 python3.7-distutils python3.8 python3.8-distutils python3.9 python3.9-distutils]: exit code: 100
```

Comment (laurentes) [reply docker/dev.dockerfile]: Sorry about this :(

Let's merge your request first, since this fixes real issues.
I will try to investigate your other issues.

Thanks for the changes!

**Review by laurentes — APPROVED** [review]

**Comment by ruomingp** [comment]
Thank you, Laurent!

**Comment by laurentes** [comment]
Heads up that this wasn't merged internally, likely because I did click on merge the changes rather than let lingvo-bot handle  things. I'm going to try to revert it and pull again. Sorry about this.

**Comment by laurentes** [comment]
Ruoming, the roll back has worked but not the roll forward, which hasn't been picked by the bot (see https://github.com/tensorflow/lingvo/pull/278).
I am going to add your fix to mine and submit a joint commit to fix the docker build (while crediting you in the description). This will hopefully unblock you.
If you really need to keep, I can subsequently roll back my changes and allow you to resend a new PR with your couple of BUILD lines changes.

Apologies for the inconvenience.

**Comment by laurentes** [comment]
With this commit: https://github.com/tensorflow/lingvo/commit/a8cbb2641c144b2ddea5726d1c23af48bf71ad2b
you should be hopefully good to go. Please let us know if you run into any other issue.
